irksome/getForm.py

```diff
@@ -124,60 +142,81 @@ def getForm(F, butch, t, dt, u0, bcs=None, bc_type="DAE"):
          either `f.interpolate(expr-c*u0)` or `f.project(expr-c*u0)`, depending
          on whether the function space for f supports interpolation or
          not.
-
     """
     v = F.arguments()[0]
     V = v.function_space()
     assert V == u0.function_space()
 
-    A = numpy.array([[Constant(aa) for aa in arow] for arow in butch.A],
-                    dtype=object)
     c = numpy.array([Constant(ci) for ci in butch.c],
                     dtype=object)
 
-    num_stages = len(c)
+    bA1, bA2 = splitting(butch.A)
+
+    try:
+        bA1inv = numpy.linalg.inv(bA1)
+    except numpy.linalg.LinAlgError:
+        bA1inv = None
+    try:
+        bA2inv = numpy.linalg.inv(bA2)
+        A2inv = numpy.array([[ConstantOrZero(aa) for aa in arow] for arow in bA2inv],
+                            dtype=object)
+    except numpy.linalg.LinAlgError:
+        raise NotImplementedError("We require A = A1 A2 with A2 invertible")
+
+    A1 = numpy.array([[Constant(aa) for aa in arow] for arow in bA1],
```

Comment:
Is there any particular reason to not use `ConstantOrZero(aa)` here?

Reply:
Good catch.  When `A = I A` you'd like to catch the sparsity.